core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -605,6 +603,29 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  def unZip(inFile: File, unzipDir: File): Unit = {
+    if (!unzipDir.mkdirs && !unzipDir.isDirectory) {
+      throw new IOException("Mkdirs failed to create " + unzipDir)
+    } else {
+      if (Shell.WINDOWS) {
+        FileUtil.unZip(inFile, unzipDir)
+      } else {
+        val untarCommand = new StringBuilder
+        untarCommand.append(s"mkdir $unzipDir && unzip -d ")
```

Comment:
Why do we need to mkdir dest directory again?

Reply:
I didn't create a new folder at the beginning, but the file is missing a level-1 directory, which is inconsistent with the original directory layer after decompression, so here is to create a folder first and then extract it.